src/v/cluster/members_backend.cc

```diff
@@ -1181,6 +1205,78 @@ void members_backend::handle_reallocation_finished(model::node_id id) {
     });
 }
 
+ss::future<> members_backend::reconcile_raft0_updates() {
+    while (!_as.local().abort_requested()) {
+        co_await _new_updates.wait([this] { return !_raft0_updates.empty(); });
+
+        auto update = _raft0_updates.front();
```

Comment:
looks like undefined behavior since i'm pretty sure the predicate on the condition variable is not guaranteed to be true when the coroutine resumes. if you look at the fast path for do_until it's true, but looks like that property does not hold for when the predicate is not initially true.

Reply:
cc @michael-redpanda